src/app/auth/guards/authorized.guard.ts

```diff
@@ -1,8 +1,27 @@
 import { Injectable } from '@angular/core';
+import { CanLoad, Router, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from '@app/auth/services/auth.service';
+
+type GuardResult = boolean | UrlTree;
+type MaybeAsync<T> = T | Observable<T> | Promise<T>;
 
 @Injectable({
   providedIn: 'root',
 })
-export class AuthorizedGuard {
+export class AuthorizedGuard implements CanLoad {
+  constructor(
+    private authService: AuthService,
+    private router: Router
+  ) {}
+
   // Add your code here
+  canLoad(): MaybeAsync<GuardResult> {
```

Comment:
I do adore the idea to declare a new type (that overlaps with the interface), but it is really not necessary.
![image](https://github.com/user-attachments/assets/4c232d06-5f1c-4834-93b7-e632b4df4213)